### PR TITLE
Moved ospool-cm to el9, fix for HTCondor API localhost issue (HTCONDOR-2056)

### PR DESCRIPTION
**Pull request by rynge**

### CODE REVIEW

File: opensciencegrid/ospool-cm/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,22 @@
 # Specify the opensciencegrid/software-base image tag
-ARG BASE_OSG_SERIES=3.6
+ARG BASE_OSG_SERIES=23
 ARG BASE_YUM_REPO=release
+ARG BASE_OS=el9
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/Dockerfile:4]: This just sets the default for the build arg if it's not passed at build time

**Review by brianhlin — COMMENTED** [review]
To be clear, this will create images with the following `OSG series, OS` based on our GHA:

- 3.6, EL7
- 23, EL9

It looks like the former may not work because of missing prometheus client packages

File: opensciencegrid/ospool-cm/Dockerfile [hunk 2]
```diff
@@ -27,6 +30,8 @@ RUN if [[ $BASE_YUM_REPO = release ]]; then \
       yum -y install condor; \
     fi
 
+RUN python3 -m pip install prometheus_client
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/Dockerfile:33]:
```suggestion
# FIXME: when we drop EL7 for this image,
#  change this back to a python3-prometheus_client RPM install
RUN python3 -m pip install prometheus_client
```

**Comment by rynge** [comment]
Yeah my preference would be el8 for 3.6, but this will work. Thanks!

**Review by brianhlin — APPROVED** [review]
LGTM but we should hold off on merging until after the UC networking issue is resolved

**Comment by brianhlin** [comment]
UC networking looks good. Merging

**Pull request merged**